### PR TITLE
feat(executorSMTP)

### PR DESCRIPTION
**Pull request by yesnault**
Signed-off-by: Yvonnick Esnault <yvonnick.esnault@corp.ovh.com>

### CODE REVIEW

**Comment by fsamin** [comment]
Hi, 
What is the use case ?

**Comment by yesnault** [comment]
It will be usefull for post step execution *if failure*

**Comment by yesnault** [comment]
And will be usefull too with pop executor (for test content of of mail, for a dry test connection)

**Comment by sguiheux** [comment]
post step execution? You mean send mail to user if failure ? This is not the aim of venom to do that. It's not a CI engine :).

**Comment by yesnault** [comment]
Ok. So it's for testing mail, with pop executor soon.

**Comment by fsamin** [comment]
Can we create issues for all executors we want to implement ? it will be easier to discuss about it

**Comment by yesnault** [comment]
Yep, 1 issue per executor

**Comment by yesnault** [comment]
I'll rebase this PR with master, your fix (for nil map) is in this pr too.

**Comment by yesnault** [comment]
@fsamin rebase done.

File: templater.go [hunk 1]
```diff
@@ -14,9 +14,6 @@ type Templater struct {
 }
 
 func newTemplater(values map[string]string) *Templater {
-	if values == nil {
```
Comment (fsamin) [inline templater.go]: why remove this ?

Comment (yesnault) [inline templater.go]: already in  Add ? We keep both ?

File: executors/smtp/executor.go [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+	if err := c.Rcpt(mailTo.Address); err != nil {
+		return fmt.Errorf("Error with c.Rcpt:%s", err.Error())
+	}
+
+	w, err := c.Data()
+	if err != nil {
+		return fmt.Errorf("Error with c.Data:%s", err.Error())
+	}
+
+	_, err = w.Write([]byte(message))
```
Comment (fsamin) [inline executors/smtp/executor.go]: one line

Comment (yesnault) [inline executors/smtp/executor.go]: :-D

**Comment by yesnault** [comment]
@sguiheux @fsamin use case added to readme

**Pull request merged**